controllers/workspace/devworkspace_controller.go

```diff
@@ -668,19 +669,23 @@ func (r *DevWorkspaceReconciler) getWorkspaceId(ctx context.Context, workspace *
 }
 
 func (r *DevWorkspaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	setupHttpClients()
-
+	setupHttpClients(mgr.GetClient())
 	maxConcurrentReconciles, err := wkspConfig.GetMaxConcurrentReconciles()
 	if err != nil {
 		return err
 	}
 
-	var emptyMapper = func(obj client.Object) []reconcile.Request {
+	var certificatesHandler = func(obj client.Object) []reconcile.Request {
+		certs, ok := readCertificates(mgr.GetClient())
+		if ok {
+			injectCertificates(certs, httpClient.Transport.(*http.Transport))
+		}
```

Comment:
Could we print logs if something went wrong?

Reply:
done